### DISCUSSION TITLE
add a way to merge two services as duplicates

### DISCUSSION THREAD

**Discussion by michielbdejong** [tosdr/edit.tosdr.org#1041] [Ideas]
For instance https://edit.tosdr.org/services/507 is a duplicate of https://edit.tosdr.org/services/347
```sh
phoenix_development=# select * from services where "id" in (347, 507);
 id  |  name   |         url          |         created_at         |         updated_at         | wikipedia | keywords | related |  slug
-----+---------+----------------------+----------------------------+----------------------------+-----------+----------+---------+---------
 347 | godaddy | godaddy              | 2018-05-08 12:19:42.984886 | 2018-05-15 09:22:14.239463 |           |          |         | godaddy
 507 | GoDaddy | https://godaddy.com/ | 2018-05-29 14:20:55.929566 | 2018-05-29 14:20:55.929566 |           |          |         |
(2 rows)

**Comment by jancborchardt** [comment]
There’s more duplicates – all of them have no content of their own so could simply be deleted?
- Discord https://edit.tosdr.org/services/585
- Reddit https://edit.tosdr.org/services/622
- Signal https://edit.tosdr.org/services/1513

**Comment by JesseWeinstein** [comment]
The three above still need to be deleted/merged.

**Comment by michielbdejong** [comment]
yeah, we should implement a button to mark as duplicate, and then we can use that to properly mark these existing duplicates

**Comment by michielbdejong** [comment]
I'll do a manual merge of all current duplicate services, as follows:
* identify which service_id should survive and which one should disappear
* update service_id in documents, points, but not service_comments
* check whether there are any domain names to merge into the list of the one that stays

**Comment by michielbdejong** [comment]
I tried this out in dev with services 300 (stackexchange) and 312 (stackoverflow). The SQL updates were easy enough, but things to keep in mind:
* when doing this directly in psql, there will be no trace of it in the comments, so better to make it a Merge button.
* there will be duplicate documents after the merge. we should actually check for duplicate documents in general, also for parent services
* should run `rake service:perform_rating` afterwards (I think Heroku runs this [once every 12 hours](https://github.com/tosdr/edit.tosdr.org/blob/master/config/schedule.rb#L23))
